customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDto.java

```diff
@@ -0,0 +1,198 @@
+package no.unit.nva.customer.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import java.net.URI;
+import java.time.Instant;
+import java.util.Objects;
+import java.util.UUID;
+import no.unit.nva.customer.model.interfaces.Customer;
+import no.unit.nva.customer.model.interfaces.JsonLdSupport;
+import nva.commons.utils.JacocoGenerated;
+
+@JsonTypeName("Customer")
+public class CustomerDto implements Customer, JsonLdSupport {
+
+    private URI id;
+    private UUID identifier;
+    private Instant createdDate;
+    private Instant modifiedDate;
+    private String name;
+    private String displayName;
+    private String shortName;
+    private String archiveName;
+    private String cname;
+    private String institutionDns;
+    private String feideOrganizationId;
+    private String cristinId;
+    @JsonProperty("@context")
+    private URI context;
+
+    public CustomerDto() {
+
+    }
+
+    @Override
+    public URI getId() {
+        return id;
+    }
+
+    @Override
+    public void setId(URI id) {
+        this.id = id;
+    }
+
+    @Override
+    public UUID getIdentifier() {
+        return identifier;
```

Comment:
I think that either when it comes to DTOs the identifier should be produced by the the Id or the opposite. 
In the draft PR (https://github.com/BIBSYSDEV/nva-customer-api/pull/26/)  I have implemented the case where 

The main idea is that the identifier is generated by the Id in a DTO. A  DAO however, has an identifier as a real field. 
It could be the case also that the the real field in a DTO is the identifier and the id is generated based on the host. In that case we would need to provide to the DTO the hosting environment to calculate the ID. I suppose this is already happening in practice when we set the Id value of the DTO.

I understand that maybe this is out of scope for this fix.

Reply:
Let's have a discussion on this later. Remember for newly created Customers we don't have an id to produce identifier from.